### PR TITLE
python312Packages.free-proxy: correct src and dependencies

### PR DESCRIPTION
**Pull request by Tochiaha**
refactoring

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Tochiaha** [comment]
@SuperSandro2000

File: pkgs/development/python-modules/free-proxy/default.nix [hunk 1]
```diff
@@ -12,16 +13,16 @@ buildPythonPackage rec {
   version = "1.1.1";
   pyproject = true;
 
-  src = fetchFromGitHub {
-    owner = "jundymerk";
-    repo = "free-proxy";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-1hTOMbsL1089/yPZbAIs5OgjtEzCBlFv2hGi+u4hV/k=";
+  src = fetchurl {
+    # using fetchurl to avoid failed to find mirror error using fetchFromGitHub.
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/free-proxy/default.nix:17]: I don't understand. This is either a general fetchFromGitHub issue which doesn't warant using fetchgit here or something specific to this package. Can you explain this in a bit more detail?

Comment (Aleksanaa) [reply pkgs/development/python-modules/free-proxy/default.nix:17]: That sounds like something wrong with their network, really. It builds on hydra. https://hydra.nixos.org/build/266126852

Comment (Tochiaha) [reply pkgs/development/python-modules/free-proxy/default.nix:17]: so the output of the current build is cpe. that is an error from me the error is based on hash value. but when i try to change the hash value i get "error: cannot download source from any mirror"

Comment (Aleksanaa) [reply pkgs/development/python-modules/free-proxy/default.nix:17]: The short answer is, yes, your hash was wrong.

```
=> nurl https://github.com/jundymek/free-proxy/archive/refs/tags/1.1.1.tar.gz
$ nix flake prefetch --extra-experimental-features 'nix-command flakes' --json github:jundymek/free-proxy/6faf092b0bd66b5004dc2dd67f8272c311c61bbb
fetchFromGitHub {
  owner = "jundymek";
  repo = "free-proxy";
  rev = "6faf092b0bd66b5004dc2dd67f8272c311c61bbb";
  hash = "sha256-n4CdkbqLLAj+54mlVOhIuK75wAHZIlSSKNRM1OgsMS0=";
}
```

But I think your error has nothing to do with `fetchFromGitHub` itself.

Comment (Tochiaha) [reply pkgs/development/python-modules/free-proxy/default.nix:17]: Thank you. You are right it has nothing to do with `fetchFromGitHub`.

File: pkgs/development/python-modules/free-proxy/default.nix [hunk 2]
```diff
@@ -13,15 +14,16 @@ buildPythonPackage rec {
   pyproject = true;
 
   src = fetchFromGitHub {
-    owner = "jundymerk";
+    owner = "jundymek";
     repo = "free-proxy";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-1hTOMbsL1089/yPZbAIs5OgjtEzCBlFv2hGi+u4hV/k=";
+    rev = "6faf092b0bd66b5004dc2dd67f8272c311c61bbb";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/free-proxy/default.nix:19]: It's just nurl. You don't have to

**Review by Aleksanaa — COMMENTED** [review]
Please reword to `python3Packages.free-proxy: correct src and dependencies`

File: pkgs/development/python-modules/free-proxy/default.nix (hunk 2 above)
Comment (Tochiaha) [reply pkgs/development/python-modules/free-proxy/default.nix]: all neccessary changes made. everyhing works as normal now.

**Comment by Aleksanaa** [comment]
It's free-proxy not freeproxy

**Comment by Aleksanaa** [comment]
Also you have to reword commit. PR title doesn't do much difference

**Pull request merged**